dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/file/batch/BatchFileOrchestrator.kt

```diff
@@ -313,40 +305,39 @@ internal class BatchFileOrchestrator(
         }
     }
 
-    private fun listSortedBatchFiles(): List<File> {
-        return rootDir.listFilesSafe(fileFilter, internalLogger).orEmpty().sorted()
+    private fun listBatchFiles(): List<File> {
+        return rootDir.listFilesSafe(fileFilter, internalLogger).orEmpty().toList()
     }
 
-    private val File.metadata: File
-        get() = File("${this.path}_metadata")
+    private fun listSortedBatchFiles(): List<File> {
+        // note: since it is using File#compareTo, lexicographical sorting will be used, meaning "10" comes before "9".
+        // but for our needs it is fine, because the moment when Unix timestamp adds one more digit will be in 2286.
```

Comment:
Few, I guess it'll be for others to deal with this then…

Reply:
Unless SDK is installed on time machine device